boards/st/stm32u5g9j_dk2/stm32u5g9j_dk2.dts

```diff
@@ -175,12 +184,19 @@
 };
 
 &i2c2 {
-	pinctrl-0 = <&i2c2_scl_pb10 &i2c2_sda_pb11>;
 	pinctrl-names = "default";
 	status = "okay";
-	clock-frequency = <I2C_BITRATE_STANDARD>;
+	clock-frequency = <I2C_BITRATE_FAST>;
+	pinctrl-0 = <&i2c2_scl_pf1 &i2c2_sda_pf0>;
```

Comment:
Keep this prop close to related prop `pinctrl-names`.
From what I checked these node was currently unused so no problem to change pin configuration, but at least you should update board doc ("Default Zephyr Peripheral Mapping")